tests/nonbonded/test_nonbonded_precomputed.py

```diff
@@ -0,0 +1,95 @@
+import jax
+
+jax.config.update("jax_enable_x64", True)
+
+import functools
+
+import jax.numpy as jnp
+import numpy as np
+import pytest
+from common import GradientTest
+
+from timemachine.lib.potentials import NonbondedPairListPrecomputed
+from timemachine.potentials import nonbonded
+
+pytestmark = [pytest.mark.memcheck]
+
+
+def test_nonbonded_precomputed_pair_list_invalid_pair_idxs():
+
+    with pytest.raises(RuntimeError) as e:
+        NonbondedPairListPrecomputed([0], [0], 2.0, 1.1).unbound_impl(np.float32)
+
+    assert "idxs.size() must be exactly 2*B" in str(e)
+
+    with pytest.raises(RuntimeError) as e:
+        NonbondedPairListPrecomputed([(0, 0)], [0.3], 2.0, 1.1).unbound_impl(np.float32)
+
+    assert "illegal pair with src == dst: 0, 0" in str(e)
+
+    with pytest.raises(RuntimeError) as e:
+        NonbondedPairListPrecomputed([(0, 1)], [0.3, 0.4], 2.0, 1.1).unbound_impl(np.float32)
+
+    assert "offset size does not match idxs size" in str(e)
+
+
+@pytest.mark.parametrize("beta", [2.0])
+@pytest.mark.parametrize("cutoff", [1.1, 10000.0])
+@pytest.mark.parametrize("precision,rtol,atol", [(np.float64, 1e-8, 1e-8), (np.float32, 1e-4, 5e-4)])
+@pytest.mark.parametrize("ixn_group_size", [4, 33, 231])
+def test_nonbonded_pair_list_precomputed_correctness(
+    ixn_group_size,
+    precision,
+    rtol,
+    atol,
+    cutoff,
+    beta,
+    example_nonbonded_potential,
+    example_conf,
+    example_box,
+    rng: np.random.Generator,
```

Comment:
nit: unused argument

Reply:
the conftest setup doesn't let me rename these - and I don't think I can just remove them.

Reply:
nevermind - I was just confused about how the conftest works. Thanks for resolving this offline.